FirebaseAppCheck/Sources/AppAttestProvider/FIRAppAttestProvider.m

```diff
@@ -118,18 +126,51 @@ - (nullable instancetype)initWithApp:(FIRApp *)app {
 #pragma mark - FIRAppCheckProvider
 
 - (void)getTokenWithCompletion:(void (^)(FIRAppCheckToken *_Nullable, NSError *_Nullable))handler {
+  [self getToken]
+      // Call the handler with the result.
+      .then(^FBLPromise *(FIRAppCheckToken *token) {
+        handler(token, nil);
+        return nil;
+      })
+      .catch(^(NSError *error) {
+        handler(nil, error);
+      });
+}
+
+- (FBLPromise<FIRAppCheckToken *> *)getToken {
+  // Check attestation state to decide on the next steps.
+  return [self attestationState].thenOn(self.queue, ^id(FIRAppAttestProviderState *attestState) {
+    switch (attestState.state) {
+      case FIRAppAttestAttestationStateUnsupported:
+        return attestState.appAttestUnsupportedError;
+        break;
+
+      case FIRAppAttestAttestationStateSupportedInitial:
```

Comment:
Is this case intended to fall through?

Reply:
Whoops, didn't click "Start Review".. more comments coming 😅

Reply:
It is, let me add a comment with a brief explanation for the future readers.

Reply:
Are you still planning to add a comment here?